pillar.example

```diff
@@ -118,26 +118,28 @@ nginx:
       # see also http://nginx.org/en/docs/example.html
       #-- - - - - -- - - -- - - - - -- - - -- - - - -- - - - - - -- - - - - - -- - - - - -- - - - - -- - - #
       config:
-        include 'snippets/letsencrypt.conf'
+        include: 'snippets/letsencrypt.conf'
         source_path: salt://path_to_nginx_conf_file/nginx.conf # IMPORTANT: This option is mutually exclusive with the rest of the
                                                           # options; if it is found other options (worker_processes: 4 and so
                                                           # on) are not processed and just upload the file from source
         worker_processes: 4
         load_module: modules/ngx_http_lua_module.so  # pass as very first in configuration; otherwise nginx will fail to start
         #pid: /var/run/nginx.pid                     # Directory location must exist (i.e. it's  /run/nginx.pid on EL7)
         events:
-          worker_connections: 768
+          worker_connections: 1024
         http:
           sendfile: 'on'
           include:
             #### Note: Syntax issues in these files generate nginx [emerg] errors on startup.  ####
             - /etc/nginx/mime.types
 
-        ### module ngx_http_log_module example
-        log_format: |-
-          main '...';
-            access_log /var/log/nginx/access_log main
-        access_log: []       #suppress default access_log option from being added
+          ### module ngx_http_log_module example
+          log_format: |-
+            main '$remote_addr - $remote_user [$time_local] "$request" '
+                 '$status $body_bytes_sent "$http_referer" '
+                 '"$http_user_agent" "$http_x_forwarded_for"';
+              access_log /var/log/nginx/access.log main;
```

Comment:
This `log_format` needs to be tested on a working setup before merging this PR.  When running through test rendering being converted to JSON, becomes:

```json
        "log_format": "main ' -  []  '\n     '   '\n     ' ';\n  access_log /var/log/nginx/access.log main;",
```

Need to ensure that the state reads it OK, though.

Reply:
Yep, I don't have a good way to test this yet so if someone could jump in and try it out, that would be appreciated.

Reply:
The state reads it OK but there are issues -- here is the result as it currently is:

```nginx
    log_format main '$remote_addr - $remote_user [$time_local] "$request" '
     '$status $body_bytes_sent "$http_referer" '
     '"$http_user_agent" "$http_x_forwarded_for"';
  access_log /var/log/nginx/access.log main;;
```

1. `access_log` line obviously erroneous.
1. Double semi-colon at the end -- basically, the final semi-colon is appended automatically, so don't need to provide in the pillar.
1. General formatting.

So an improved version fixing the above issues:

```sls
          log_format: |-
            main '$remote_addr - $remote_user [$time_local] $status '
                                '"$request" $body_bytes_sent "$http_referer" '
                                '"$http_user_agent" "$http_x_forwarded_for"'
```

Which results in:

```nginx
    log_format main '$remote_addr - $remote_user [$time_local] $status '
                    '"$request" $body_bytes_sent "$http_referer" '
                    '"$http_user_agent" "$http_x_forwarded_for"';
```

This matches examples in the upstream docs:

* The same entry as the [full example](https://www.nginx.com/resources/wiki/start/topics/examples/full/).
* The same layout as shown in both examples on the [logging](https://docs.nginx.com/nginx/admin-guide/monitoring/logging/#setting-up-the-access-log) page.